### ISSUE TITLE
enhancement: key to dump cargo

### ISSUE THREAD

**Issue by TJesionowski** [endless-sky/endless-sky#2195]
Because having to go through the player information menu in an emergency is somewhat immersion breaking...

**Comment by retrue** [comment]
And losing your cargo when accidentally touching that key is a problem too.

**Comment by Bladewood** [comment]
Add a check?

**Comment by TJesionowski** [comment]
Make it disabled by default? I agree this shouldn't be a thing you can do by accident, I just want it to be an option.

**Comment by Amazinite** [comment]
Would it just dump everything or bring up a menu? If it brings up a menu so that you can actually choose what you dump then really it's just a shortcut to the ship info.

**Comment by TJesionowski** [comment]
No, if enabled it would dump normal cargo on first press, then outfit cargo. Ideally it would only apply to the flagship, but I imagine there'd have to be some way to have it apply to escorts as well...

**Comment by zwparchman** [comment]
Would it be better to have a "take all of my cargo" command for your fleet. If it did what it said then this whole dumping cargo into another ship stuff would be over, and it would remove the need to check if the ship you want to dump cargo into full. It's just tedious.

**Comment by KiLEdEnNis** [comment]
the current system may break the immersion a bit but it avoids all accidents & you can pick what to dump, which seems fine
being able to board escorts & switch cargo seems like a good idea though

**Comment by TJesionowski** [comment]
That would be nice. It would also be a more logical way of transferring cargo out of your flagship (the current system is simply comedic) 
I think there are some circumstances where dumping cargo is more convenient, (ie:  your flagship is faster than your escorts) but your suggestion makes far more sense than literally throwing cargo across the vacuum of space.

**Comment by zwparchman** [comment]
@KiLEdEnNis When would it be desirable to keep some of the cargo? The only issue I can think of where a fleet wide "take all of my cargo" isn't satisfactory is when you want to control the mass of each ship. Then a targeted version (which already exists for all other fleet commands) would hit > 90% of those situations.

**Comment by KiLEdEnNis** [comment]
luxury goods are way more expensive than food, usually a player wants to dump the cheap commodities then

**Comment by zwparchman** [comment]
While the current system allows it, using fleet cargo capacity is a pain. Right now the best way is the following:
1) Board a ship and loot cargo
2) Find an escort with cargo space
3) Tell the escort to follow you
4) Open your player menu and dump as much cargo will fit in the escort.
5) Prepare to pick up anything the escort didn't 
6)  Repeat

You could argue steps 2 and 3 aren't necessary, but if your goal is not to lose cargo then the are unarguably better to do.

What I would like to see:
1) Board a ship and loot cargo
2) Press a button to issue a fleet command and wait for your fleet to finish the order
3) Repeat (and since you never left the target reboarding would be quick)

As you said transferring cargo in space is currently possible and it is even easy, but it is tedious. This would be a quality of life improvement. Things such as automatic landing on a planet, automatic execution of multiple jumps to distant systems, automatic weapon firing, and automatic crew rehiring (when below the required crew) are also quality of life improvements already added to the game. This would just be another.

**Comment by KiLEdEnNis** [comment]
this issue is more about adding a hotkey for dumping cargo, there's another one for cargo transferring to/from escorts

**Comment by Wrzlprnft** [comment]
the problem is the same as why escorts shall not be able to board enemies themselves: crew salaries. if we assume there are three inappropriate large fleets, one for trading, one for jobs, and one for plundering/capturing. all of them are going to make good money, but all of them have to pay crew salaries. if the capture fleet (which is the one most likely, if we ever get different crew roles, or just buy number of crew members, to pay the most) now get's to dodge that, balance goes over board. further, crew salaries are to keep fleets smaller, for the sake of the engine, as well as the one-ship/small-fleet lorebase we have.


the dumping and picking up of cargo were never introduced to transfer cargo between ships, and without the fleet controls, it probably would not be savely possible. i doubt that this is going to get easier.

**Comment by Bladewood** [comment]
*dominates every world in human space*
